src/controller.cpp

```diff
@@ -290,8 +290,7 @@ void Controller::serverUnavailable() {
 
   m_nextStep = ServerUnavailable;
 
-  if (m_state == StateOn || m_state == StateSwitching ||
-      m_state == StateSilentSwitching || m_state == StateConnecting ||
+  if (m_state == StateSwitching || m_state == StateConnecting ||
```

Comment:
Thank you for taking this on @Gela! Two things: 
- Do we know how we enter into this server unavailable routine while in `NoSignal` and should we add logic so that we _don't_ go into `serverUnavailable()` while in `NoSignal`?  
- Is there value in deactivating here at all? Perhaps we can stipulate that deactivation must be user initiated and remove the block entirely.

Reply:
> Do we know how we enter into this server unavailable routine while in NoSignal and should we add logic so that we don't go into serverUnavailable() while in NoSignal?

I'm not very clear on how/why we go into `NoSignal` and if that's intended. I'm not sure how to reproduce that scenario to debug currently.

> Is there value in deactivating here at all? Perhaps we can stipulate that deactivation must be user initiated and remove the block entirely.

If we're in StateConnecting and the server location becomes unavailable, I think it makes sense to deactivate as opposed to remaining in the Connecting... state. Similarly in StateSwitching the VPN has to deactivate to switch servers anyway. I think it's fair here to deactivate since the VPN wouldn't have been active anyway. What are your thoughts?

Reply:
To test the `NoSignal` state in the past I have done things to break the connection to the wireguard server, for example:

1. Disconnect Wi-Fi/Ethernet and then re-connect after it tries server switching. 
2. Locate the server's IP and then install a `/32` route to drop all packets to it....

  - Linux: `ip route add blackhole <server-ip>`
  - MacOS: `route -v add -net <server-ip>/32 127.0.0.1 -blackhole`